modules/scalacheck/src/main/scala/GenModule.scala

```diff
@@ -4,6 +4,37 @@ package schema
 
 package scalacheck
 
+import org.scalacheck.Gen
+
 trait GenModule extends SchemaModule {
-  // def gen[A](schema: Schema[A])(implicit arb: Arbitrary[Prim[_]]): Gen[A]
+
+  def toGen: Schema ~> Gen = new (Schema ~> Gen) {
+    override def apply[A](schema: Schema[A]): Gen[A] = schema match {
+      case _: Schema.PrimSchema[_]        => ???
+      case record: Schema.RecordSchema[_] => recordGen(record)
+      case union: Schema.Union[_]         => unionGen(union)
+      case seq: Schema.SeqSchema[_]       => seqGen(seq)
+    }
+  }
+
+  def recordGen[A](schema: Schema.RecordSchema[A]): Gen[A] = {
+    schema.terms.map {
+      case essential: Schema.Field.Essential[_, _]       => toGen(essential.base)
+      case nonEssential: Schema.Field.NonEssential[_, _] => Gen.option(toGen(nonEssential.base))
+    }
+
+    ???
```

Comment:
I'm not sure how to generate `Gen` that produces objects of type `A`  here from list of field generators that are defined in `RecordSchema` for `A`. @vil1 do you have any suggestions?